### PR TITLE
enable rsa feature for libp2p

### PR DESCRIPTION
**Pull request by LesnyRumcajs**
## Summary of changes

Changes introduced in this pull request:

- Forest fails to connect to more strict bootstrap peers. This change permits upgrading the connections to encrypted ones (to my understanding):
```
2024-02-27T12:11:47.625785Z DEBUG Transport::dial{address=/dns4/bootstrap-venus.mainnet.filincubator.com/tcp/8888/p2p/QmQu8C6deXwKvJP2D8B6QGyhngc3ZiDnFzEHBDx8yeBXST}: multistream_select::dialer_select: Dialer: Received confirmation for protocol: /noise
2024-02-27T12:11:47.894584Z DEBUG Transport::dial{address=/dns4/bootstrap-venus.mainnet.filincubator.com/tcp/8888/p2p/QmQu8C6deXwKvJP2D8B6QGyhngc3ZiDnFzEHBDx8yeBXST}: libp2p_identity::keypair: support for RSA was disabled at compile-time
2024-02-27T12:11:47.894605Z DEBUG Transport::dial{address=/dns4/bootstrap-venus.mainnet.filincubator.com/tcp/8888/p2p/QmQu8C6deXwKvJP2D8B6QGyhngc3ZiDnFzEHBDx8yeBXST}: libp2p_core::upgrade::apply: Failed to upgrade outbound stream upgrade=/noise
2024-02-27T12:11:47.894638Z DEBUG Swarm::poll: libp2p_kad::behaviour: Last remaining address of peer is unreachable. peer=QmQu8C6deXwKvJP2D8B6QGyhngc3ZiDnFzEHBDx8yeBXST address=/dns4/bootstrap-venus.mainnet.filincubator.com/tcp/8888/p2p/QmQu8C6deXwKvJP2D8B6QGyhngc3ZiDnFzEHBDx8yeBXST
2024-02-27T12:11:47.894652Z DEBUG Swarm::poll: libp2p_swarm: Connection attempt to peer failed with Transport([("/dns4/bootstrap-venus.mainnet.filincubator.com/tcp/8888/p2p/QmQu8C6deXwKvJP2D8B6QGyhngc3ZiDnFzEHBDx8yeBXST", Other(Custom { kind: Other, error: Other(Left(Right(Apply(InvalidKey(DecodingError { msg: "cargo feature `rsa` is not enabled", source: None }))))) }))]). peer=QmQu8C6deXwKvJP2D8B6QGyhngc3ZiDnFzEHBDx8yeBXST
```

## Reference issue to close (if applicable)

Closes https://github.com/ChainSafe/forest/issues/3990

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -118,6 +118,7 @@ libp2p = { version = "0.53", default-features = false, features = [
   'tokio',
   'macros',
   'serde',
+  'rsa',
```
Comment (hanabi1224) [inline Cargo.toml:121]: Should we explicitly enable all of `secp256k1`, `ecdsa`, `rsa` and `ed25519`?

**Review by hanabi1224 — APPROVED** [review]

**Review by lemmih — APPROVED** [review]
What does enabling `rsa` do?

**Comment by LesnyRumcajs** [comment]
> What does enabling `rsa` do?

```
2024-02-27T12:11:47.625785Z DEBUG Transport::dial{address=/dns4/bootstrap-venus.mainnet.filincubator.com/tcp/8888/p2p/QmQu8C6deXwKvJP2D8B6QGyhngc3ZiDnFzEHBDx8yeBXST}: multistream_select::dialer_select: Dialer: Received confirmation for protocol: /noise
2024-02-27T12:11:47.894584Z DEBUG Transport::dial{address=/dns4/bootstrap-venus.mainnet.filincubator.com/tcp/8888/p2p/QmQu8C6deXwKvJP2D8B6QGyhngc3ZiDnFzEHBDx8yeBXST}: libp2p_identity::keypair: support for RSA was disabled at compile-time
2024-02-27T12:11:47.894605Z DEBUG Transport::dial{address=/dns4/bootstrap-venus.mainnet.filincubator.com/tcp/8888/p2p/QmQu8C6deXwKvJP2D8B6QGyhngc3ZiDnFzEHBDx8yeBXST}: libp2p_core::upgrade::apply: Failed to upgrade outbound stream upgrade=/noise
2024-02-27T12:11:47.894638Z DEBUG Swarm::poll: libp2p_kad::behaviour: Last remaining address of peer is unreachable. peer=QmQu8C6deXwKvJP2D8B6QGyhngc3ZiDnFzEHBDx8yeBXST address=/dns4/bootstrap-venus.mainnet.filincubator.com/tcp/8888/p2p/QmQu8C6deXwKvJP2D8B6QGyhngc3ZiDnFzEHBDx8yeBXST
2024-02-27T12:11:47.894652Z DEBUG Swarm::poll: libp2p_swarm: Connection attempt to peer failed with Transport([("/dns4/bootstrap-venus.mainnet.filincubator.com/tcp/8888/p2p/QmQu8C6deXwKvJP2D8B6QGyhngc3ZiDnFzEHBDx8yeBXST", Other(Custom { kind: Other, error: Other(Left(Right(Apply(InvalidKey(DecodingError { msg: "cargo feature `rsa` is not enabled", source: None }))))) }))]). peer=QmQu8C6deXwKvJP2D8B6QGyhngc3ZiDnFzEHBDx8yeBXST
```
To my understanding, it allows upgrading the connection to an encrypted one.

File: Cargo.toml (hunk 1 above)
Comment (LesnyRumcajs) [reply Cargo.toml:121]: Let's add them. I'll ask around to ensure it's ultimately what we need.

Comment (hanabi1224) [reply Cargo.toml:121]: As per the code, the list should be exhaustive
```rust
#[derive(Debug, Clone)]
#[allow(clippy::large_enum_variant)]
enum KeyPairInner {
    /// An Ed25519 keypair.
    #[cfg(feature = "ed25519")]
    Ed25519(ed25519::Keypair),
    /// An RSA keypair.
    #[cfg(all(feature = "rsa", not(target_arch = "wasm32")))]
    Rsa(rsa::Keypair),
    /// A Secp256k1 keypair.
    #[cfg(feature = "secp256k1")]
    Secp256k1(secp256k1::Keypair),
    /// An ECDSA keypair.
    #[cfg(feature = "ecdsa")]
    Ecdsa(ecdsa::Keypair),
}
```

**Review by hanabi1224 — APPROVED** [review]

**Pull request merged**